winlogbeat/beater/eventlogger.go

```diff
@@ -171,6 +171,9 @@ runLoop:
 			e.log.Debugf("Read() returned %d records.", len(records))
 			if len(records) == 0 {
 				time.Sleep(time.Second)
+				if stop {
```

Comment:
We've got the right idea here, but unfortunately this will only work if `len(records) == 0`. It is possible for an implementation of `eventlog.EventLog` to return records with an `io.EOF` (see `Read()` in `wineventlog_experimental.go`). What I think will then happen in this case is that it will break out of the inner loop and then get stuck in the outer loop (similar to the original bug). It may not do this indefinitely, but it could append additional logs to the output file.

I think what needs to happen here is that the stop variable needs to be checked in the outer loop as well:

```go
runLoop:
	for stop := false; !stop; {
		err = api.Open(state)
		// ...

		for !stop {
			// ...
			case io.EOF:
				// Graceful stop.
				stop = true
			// ...
		}
	}
```